projects/packages/my-jetpack/src/class-rest-products.php

```diff
@@ -17,32 +17,77 @@ class REST_Products {
 	public function __construct() {
 		register_rest_route(
 			'my-jetpack/v1',
-			'/site/products',
+			'site/products',
 			array(
-				'methods'             => \WP_REST_Server::READABLE,
-				'callback'            => __CLASS__ . '::get_products',
-				'permission_callback' => __CLASS__ . '::permissions_callback',
+				array(
+					'methods'             => \WP_REST_Server::READABLE,
+					'callback'            => __CLASS__ . '::get_products',
+					'permission_callback' => __CLASS__ . '::permissions_callback',
+				),
+				'schema' => array( $this, 'get_products_schema' ),
 			)
 		);
 
+		$product_arg = array(
+			'description'       => __( 'Product slug', 'jetpack-my-jetpack' ),
+			'type'              => 'string',
+			'enum'              => Products::get_product_names(),
+			'required'          => false,
+			'validate_callback' => __CLASS__ . '::check_product_argument',
+		);
+
 		register_rest_route(
-			'my-jetpack/v1/',
-			'/site/products/(?P<product>[a-z\-]+)',
+			'my-jetpack/v1',
+			'site/products/(?P<product>[a-z\-]+)',
 			array(
-				'methods'             => \WP_REST_Server::READABLE,
-				'callback'            => __CLASS__ . '::get_product',
-				'permission_callback' => __CLASS__ . '::permissions_callback',
-				'args'                => array(
-					'product' => array(
-						'description'       => __( 'Product slug', 'jetpack-my-jetpack' ),
-						'type'              => 'string',
-						'enum'              => Products::get_product_names(),
```

Comment:
Why remove the `enum`? it will check the product name

Reply:
it's there on line 31 ;)